src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_ARM64/ARM64Emitter.cs

```diff
@@ -63,7 +63,7 @@ public void EmitLDR(Register regDst, Register regAddr)
 
         public void EmitLDR(Register regDst, Register regSrc, int offset)
         {
-            Debug.Assert(offset >= -255 && offset <= 4095);
+            Debug.Assert(offset >= -255 && offset <= 32760);
```

Comment:
We may want to `throw new NotImplementedException();` when the offset is too big. I think it is possible to construct cases where the offset is too big. It is better to throw exception than emit bad code silently.